### DISCUSSION TITLE
How can I map Romanian Source `grave_accent_and_tilde` to an actual GRAVE ACCENT / TILDE?

### DISCUSSION THREAD

**Discussion by ghost** [pqrs-org/Karabiner-Elements#3965] [Q&A]
In MacOS Keyboard Settings, I use two sources: British, and Romanian - Standard.
<img width="222" alt="Screenshot 2024-10-10 at 16 07 49" src="https://github.com/user-attachments/assets/4084fb99-69c6-426c-9a5e-0904169b0aeb">

I have a Simple Modification in Karabiner that maps `non_us_backslash` to `grave_accent_and_tilde`. This works well when the British Source is selected.
<img width="547" alt="Screenshot 2024-10-10 at 16 08 46" src="https://github.com/user-attachments/assets/f04ebf1d-6259-4934-a1cd-34ee50b9b0d1">

However, when the Romanian- Standard source is selected, I get `\\||`, when pressing the same key. I believe this is because MacOS interprets `grave_accent_and_tilde` as that specific backslash/bar character.
<img width="322" alt="Screenshot 2024-10-10 at 16 14 06" src="https://github.com/user-attachments/assets/aad1efe7-9dd7-4de5-935a-02d9bb81b061">

```
The following key presses:
[
  {
    "type": "down",
    "name": {"key_code":"grave_accent_and_tilde"},
    "usagePage": "7 (0x0007)",
    "usage": "53 (0x0035)",
    "misc": ""
  },
  {
    "type": "up",
    "name": {"key_code":"grave_accent_and_tilde"},
    "usagePage": "7 (0x0007)",
    "usage": "53 (0x0035)",
    "misc": ""
  },
  {
    "type": "down",
    "name": {"key_code":"right_shift"},
    "usagePage": "7 (0x0007)",
    "usage": "229 (0x00e5)",
    "misc": "flags right_shift"
  },
  {
    "type": "down",
    "name": {"key_code":"grave_accent_and_tilde"},
    "usagePage": "7 (0x0007)",
    "usage": "53 (0x0035)",
    "misc": "flags right_shift"
  },
  {
    "type": "up",
    "name": {"key_code":"grave_accent_and_tilde"},
    "usagePage": "7 (0x0007)",
    "usage": "53 (0x0035)",
    "misc": "flags right_shift"
  },
  {
    "type": "up",
    "name": {"key_code":"right_shift"},
    "usagePage": "7 (0x0007)",
    "usage": "229 (0x00e5)",
    "misc": ""
  }
]

Output the following string:
\\||

Desired string:
``~~
```
---

How can I map the `grave_accent_and_tilde` key to actually output a grave accent or a tilde while the Romanian - Standard source is selected as well?

**Comment by gforien** [comment]
You'd have to write a [complex modification](https://karabiner-elements.pqrs.org/docs/json/typical-complex-modifications-examples/) in JSON for this

Complex modifications allow to define conditions, like in your example
- map `non_us_backslash` to `grave_accent_and_tilde` if `language` is `en`
- map `backslash` to `grave_accent_and_tilde` if `language` is `ro`

This can be done with the [`input_source_if`](https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/conditions/input-source/) condition operator

If you're new to complex modifications you may find useful the collection of predefined rules (https://ke-complex-modifications.pqrs.org/) or this generator (https://genesy.github.io/karabiner-complex-rules-generator/)
